File: 345/Solution.java
class Solution {
    public String reverseVowels(String s) {
        char []x  = s.toCharArray();
        int i = 0;
        int j = x.length - 1;
        String vowels = "aeiouAEIOU";
        while(i<j) {
            if(vowels.contains(x[i]+"") && vowels.contains(x[j]+"")) {
                char temp = x[i];
                x[i] = x[j];
                x[j] = temp;
                i++;
                j--;
            }
            else if (vowels.contains(x[i]+"")) {
                j--;
            }
            else if (vowels.contains(x[j]+"")){
                i++;
            }
            else {
                i++;
                j--;
            }
                        
        }
        return new String(x);
    }
}

File: 345/Solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        x = [l for l in s]
        i=0
        j=len(x)-1
        
        vowels = 'aeiouAEIOU'
        
        while i<j:
            if x[i] in vowels and x[j] in vowels:
                temp = x[i]
                x[i] = x[j]
                x[j] = temp
                i+=1
                j-=1
            elif x[i] in vowels:
                j-=1
            elif x[j] in vowels:
                i+=1
            else:
                i+=1
                j-=1
        
        return ''.join(x)